src/Connect.js

```diff
@@ -364,8 +354,9 @@ class Connect {
         this._state = { ...this._state, ...update }
         break
       case 'function':
-        this._state = { ...this._state, ...update(this._state) }
+        this._state = update(this._state)
         break
+        // TODO may want not partially apply, or move this portion to another function
```

Comment:
should we distinguish maybe between `setState` and something like `replaceState`? As it is, the functions passed to setState behave differently than objects, which could be confusing, though we do need a way of completely writing over the state.

Reply:
yeah possibly, I think the function and and object partially applied are similar, the object be like a default state function, while this one feels different. I was considering that maybe loadState could now be a place for it